src/AuthenticationRequestParameters.ts

```diff
@@ -41,20 +41,20 @@ export class AuthenticationRequestParameters {
   extraQueryParameters: string;
   loginHint: string;
   domainHint: string;
-  redirectUri: string;
+  redirectUri: string;  
   public get authority(): string {
     return this.authorityInstance.CanonicalAuthority;
   }
 
-  constructor(authority: Authority, clientId: string, scope: Array<string>, responseType: string, redirectUri: string) {
+  constructor(authority: Authority, clientId: string, scope: Array<string>, responseType: string, redirectUri: string, state: string = "") {
     this.authorityInstance = authority;
     this.clientId = clientId;
     this.scopes = scope;
     this.responseType = responseType;
     this.redirectUri = redirectUri;
     // randomly generated values
     this.correlationId = Utils.createNewGuid();
-    this.state = Utils.createNewGuid();
+    this.state = state && state != "" ? state : Utils.createNewGuid();
```

Comment:
We can't do this. A new state value needs to be generated every time to prevent CSRF attack. We should append Utils.createNewGuid() with the state value passed . Later when we return the state value back to you, we will have to strip it down.